cookiebanner/templates/cookiebanner/bootstrap4.html

```diff
@@ -59,26 +59,25 @@ <h4 class="cookiebannerH4">{{ cookiegroup.name }}</h4>
   const cookiegroups = JSON.parse("{{ cookiegroups_json|escapejs }}");
 
   document.addEventListener("DOMContentLoaded", function () {
-    let keyValue = document.cookie.match('(^|;) ?cookiebanner=([^;]*)(;|$)');
-    let cookiebannerCookie = keyValue ? decodeURIComponent(keyValue[2]) : null;
+    let keyValue = document.cookie.split(/;\s*/g).find((x) => x.startsWith("cookiebanner"));
+    let cookiebannerCookie = keyValue ? decodeURIComponent(keyValue.slice(13)) : null;
     if (cookiebannerCookie) return;
 
-    $('#cookiebannerModal').modal({
+    $?.('#cookiebannerModal').modal?.({
```

Comment:
Will this work in older browsers? I'm thinking `$?.` might not be supported by older ones

Reply:
According to [this](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining#browser_compatibility), it should be compatible with all major browsers except Internet Explorer. However, it is a very recent feature.

I have seen another solution using `if (typeof foo.functionName === "function")`. Would something like that be preferable?